### DISCUSSION TITLE
Fresh Ruby bindings for tree-sitter

### DISCUSSION THREAD

**Discussion by stackmystack** [tree-sitter/tree-sitter#1841] [Bindings]
Hello all,

I'm the author of the newly-written `ruby` bindings for `tree-sitter`, and I would like to share with the community the progress we've done. Here's the link: https://github.com/Faveod/ruby-tree-sitter .

I don't know if you're aware of the current situation, but the `ruby` bindings you're linking to in your website, and which is under [`github.com/tree-sitter`](https://github.com/tree-sitter/ruby-tree-sitter), are long obsolete, unmaintained, and the authors seem to be unresponsive. There's even [an issue here in this repo](https://github.com/tree-sitter/tree-sitter/issues/1752) mentioning it.

Our bindings have been tested rigorously in our production environment, and this usage feeds back into their maintenance. As an example, we found a very peculiar `SEGFAULT` that was triggered in our environment, and we managed to fix and publish it. I've written about it in detail in our [docs](https://github.com/Faveod/ruby-tree-sitter/blob/master/docs/SIGSEGV.md).

We also extensively test the quality of our work by invoking `valgrind` and `asan` on our test suite which already cover almost all the bindings.

Can you consider the new bindings as replacement to the existing ones?

I would happily present myself as the maintainer since I am already doing all the necessary work.

**Comment by maxbrunsfeld** [comment]
This is awesome! I definitely think we should replace the current ruby bindings with this. Would you like to move this repo into the Tree-sitter org?

**Reply by stackmystack** [reply]
Yes I'd love to. How can we proceed? The old bindings are `ruby-tree-sitter` and our repo is called the same, so I guess there would be a clash there, right?

**Reply by stackmystack** [reply]
@maxbrunsfeld When can we move the repo? I'm ready whenever you are.

**Reply by akimd** [reply]
@maxbrunsfeld Is there something that prevents this from happening?
